### PR TITLE
Product tests launcher CONTAINER_TEMPTO_PROFILE_CONFIG related cleanup

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hadoop.java [hunk 1]
```diff
@@ -82,8 +84,8 @@ public void extendEnvironment(Environment.Builder builder)
             portBinder.exposePort(container, 50075); // DataNode Web UI prior to Hadoop 3
         });
 
-        builder.configureContainer(
-                COORDINATOR,
+        builder.configureContainersIfPresent(
```
Comment (wendigo) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hadoop.java:87]: I'd reuse existing code:

`builder.configureContainers(container -> if (isPrestoContainer(container.getLogicalName()) {`

Comment (kokosing) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hadoop.java:87]: That was my plan, but then diff was much bigger and it was not clear what changed so introduced a new method. Also I think it is ok to wrap such common use cases as dedicated methods.

**Review by wendigo — APPROVED** [review]
LGTM.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java [hunk 2]
```diff
@@ -34,4 +38,19 @@ public static boolean isPrestoContainer(String name)
...
         return name.startsWith(PRESTO);
     }
+
+    public static void configureTempto(Environment.Builder builder, DockerFiles.ResourceProvider configDir, String fileDescription)
+    {
+        builder.configureContainer(TESTS, dockerContainer -> {
+            String temptoConfig = "/docker/presto-product-tests/conf/tempto/tempto-configuration-for-" + fileDescription + ".yaml";
+            dockerContainer
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("tempto-configuration.yaml")),
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java:48]: expecting `{configDir}/tempto-configuration.yaml` isn't very flexible.

we have many `tempto-configuration*.yaml` files and only some of them match this convention

```
$ find ./testing/trino-product-tests-launcher/src -name 'tempto-configuration*.yaml' | sed 's@.*/@@'
tempto-configuration.yaml
tempto-configuration.yaml
tempto-configuration-for-docker-tls.yaml
tempto-configuration-for-docker-default.yaml
tempto-configuration-for-docker-ldap.yaml
tempto-configuration-for-docker-oauth2.yaml
tempto-configuration-for-hive3.yaml
tempto-configuration-profile-config-file.yaml
tempto-configuration.yaml
tempto-configuration.yaml
tempto-configuration.yaml
```

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeLdapAndFile.java [hunk 3]
```diff
@@ -59,17 +59,7 @@ public void extendEnvironment(Environment.Builder builder)
                             CONTAINER_PRESTO_ETC + "/password.db");
         });
 
-        builder.configureContainer(TESTS, dockerContainer -> {
-            String temptoConfig = "/docker/presto-product-tests/conf/tempto/tempto-configuration-for-singlenode_ldap_and_file.yaml";
-            dockerContainer
-                    .withCopyFileToContainer(
-                            forHostPath(configDir.getPath("tempto-configuration.yaml")),
-                            temptoConfig)
-                    .withEnv("TEMPTO_CONFIG_FILES", temptoConfigFiles ->
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeLdapAndFile.java:68]: Why this env uses TEMPTO_CONFIG_FILES env, while eg MulinodeKafka binds directly to `tempto-configuration-profile-config-file.yaml`?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HadoopKerberos.java [hunk 4]
```diff
@@ -26,8 +26,8 @@
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.HADOOP;
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.configureTempto;
 import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_CONFIG_PROPERTIES;
-import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TEMPTO_PROFILE_CONFIG;
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HadoopKerberos.java:30]: > Use configureTempto instead CONTAINER_TEMPTO_PROFILE_CONFIG

- Update MultinodeKafka too (and maybe others)
- deprecate `CONTAINER_TEMPTO_PROFILE_CONFIG` with suggestion what to use instead

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HadoopKerberos.java [hunk 5]
```diff
@@ -73,8 +73,8 @@ public void extendEnvironment(Environment.Builder builder)
         });
         builder.configureContainer(TESTS, container -> {
             container.setDockerImageName(dockerImageName);
-            container.withCopyFileToContainer(forHostPath(configDir.getPath("tempto-configuration.yaml")), CONTAINER_TEMPTO_PROFILE_CONFIG);
         });
+        configureTempto(builder, configDir, "hadoop-kerberos");
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HadoopKerberos.java:77]: this filedescription is nice, but it's not copy-paste-friendly.  would you consider dropping it?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hadoop.java [hunk 6]
```diff
@@ -66,8 +68,8 @@ public void extendEnvironment(Environment.Builder builder)
     {
         builder.addContainer(createHadoopContainer(dockerFiles, portBinder, hadoopBaseImage + ":" + hadoopImagesVersion, HADOOP));
 
-        builder.configureContainer(
-                COORDINATOR,
+        builder.configureContainersIfPresent(
+                ImmutableSet.of(COORDINATOR, WORKER),
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hadoop.java:72]: I like the syntax, but there is one downside:

- we know COORDINATOR _should_ exists, so it's actually an error if it doesn't
- we want to apply this is conditionally to WORKER, if it exists

maybe 

```
configureContainers(Set<String> containersToConfigure, Set<String> optionalContainersToConfigure, Consumer)
```

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java [hunk 7]
```diff
@@ -13,6 +13,7 @@
  */
 package io.trino.tests.product.launcher.env.common;
 
+import com.google.common.collect.ImmutableSet;
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java:16]: >  Make HydraIdentityProvider env extender Trino nodes friendly


maybe cmt msg "Configure worker if present in HydraIdentityProvider env extender"

**Review by findepi — APPROVED** [review]

**Review by kokosing — COMMENTED** [review]
I extracted https://github.com/trinodb/trino/pull/9090 out of this PR. I have some issues with TLS environment which is not clear to me why it happens.

**Pull request closed without merging**